docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/set-metadata.md

```diff
@@ -0,0 +1,121 @@
+---
+sidebar_label: set-metadata
+description: Updates metadata on Stage or Freight resources during the promotion process.
+---
+
+# `set-metadata`
+
+`set-metadata` updates metadata on `Stage` or `Freight` resources during the
+promotion process. This step allows you to attach arbitrary key-value pairs to the
```

Comment:
Nit: I think this is the more conventional way to denote key/value:

```suggestion
promotion process. This step allows you to attach arbitrary key/value pairs to the
```

Reply:
I found "key-value" in a lot of spots, but probably missed some. Please be on the lookout for it.